tests/mir-opt/const_array_locals.rs

```diff
@@ -0,0 +1,46 @@
+//@ test-mir-pass: GVN
+#![feature(repr_simd)]
+
+// #[repr(simd)]
+// struct F32x8([f32; 8]);
+
+// EMIT_MIR const_array_locals.main.GVN.diff
+// CHECK-LABEL: fn main(
+#[rustfmt::skip]
+pub fn main() {
+    // CHECK: let [[array_lit:_.*]]: [i32; 32];
+    // CHECK:     debug _arr => [[arr:_.*]];
+
+    let _arr = [
+        255, 105, 15, 39, 62, 251, 191, 178, 9, 4, 56, 221,
+        193, 164, 194, 197, 6, 243, 218, 171, 87, 247, 104,
+        159, 22, 157, 105, 31, 96, 173, 50, 1,
+    ];
+    let _barr = [
+        255, 105, 15, 39, 62, 251, 191, 178, 9, 4, 56, 221,
+        193, 164, 194, 197, 6, 243, 218, 171, 87, 247, 104,
+        159, 22, 157, 105, 31, 96, 173, 50, 1,
+    ];
+    // let _foo = [
+    //     [255, 105, 15, 39, 62, 251, 191, 178, 9, 4, 56, 221],
+    //     [193, 164, 194, 197, 6, 243, 218, 171, 87, 247, 104, 42],
+    // ];
+    let _darr = *&[
```

Comment:
This returns `Some(OpTy { op: Indirect(MemPlace { ptr: alloc2<imm>, meta: None, misaligned: None }), ty: [i32; 32] })`